packages/vhd-lib/src/Vhd/VhdDirectory.js

```diff
@@ -174,12 +174,14 @@ export class VhdDirectory extends VhdAbstract {
   // and if the full block is modified in child ( which is the case whit xcp)
 
   async coalesceBlock(child, blockId) {
-    assert.strictEqual(this._handler, child._handler)
+    if (!(child instanceof VhdDirectory) || this._handler !== child._handler) {
+      return super.coalesceBlock(child, blockId)
+    }
     await this._handler.copy(
       child.getChunkPath(child._getBlockPath(blockId)),
       this.getChunkPath(this._getBlockPath(blockId))
     )
-    return await this._handler.getSize(this.getChunkPath(this._getBlockPath(blockId)))
+    return sectorsToBytes(this.sectorsPerBlock)
```

Comment:
Feel free to add a `VhdAbstract#_blockSize` getter if this value is often needed.

Reply:
I think it should be in another PR, but I don't really want to push backs this one

Reply:
Fair enough :+1: